src/xdr_rdma.c

```diff
@@ -579,14 +619,27 @@ xdr_rdma_post_send_n(RDMAXPRT *xprt, struct rpc_rdma_cbc *cbc, int sge,
 	ret = ibv_post_send(xprt->qp, &cbc->wr.wwr, &xprt->bad_send_wr);
 	if (ret) {
 		__warnx(TIRPC_DEBUG_FLAG_ERROR,
-			"%s() %p[%u] cbc %p ibv_post_send failed: %s (%d)",
-			__func__, xprt, xprt->state, cbc, strerror(ret), ret);
+			"%s() %p[%u] cbc %p ibv_post_send failed: %s (%d), "
+			"num sge %d imm_data %d",
+			__func__, xprt, xprt->state, cbc, strerror(ret), ret,
+			cbc->wr.wwr.num_sge, cbc->wr.wwr.imm_data);
 		return ret; // FIXME np_uerror(ret)
 	}
 
 	return 0;
 }
 
+static inline int
+xdr_rdma_wait_cb_done_locked(struct rpc_rdma_cbc *cbc)
+{
+	struct timespec ts;
+	ts.tv_sec = time(NULL) + RDMA_CB_TIMEOUT_SEC;
```

Comment:
Waiting for 60 seconds with a lock held seems like a bad idea.  Is there no way to drop the lock around the wait?